ddl/column.go

```diff
@@ -631,15 +631,6 @@ func checkDropColumn(t *meta.Meta, job *model.Job) (*model.TableInfo, *model.Col
 		return nil, nil, nil, errors.Trace(err)
 	}
 	idxInfos := listIndicesWithColumn(colName.L, tblInfo.Indices)
-	if len(idxInfos) > 0 {
-		for _, idxInfo := range idxInfos {
-			err = checkDropIndexOnAutoIncrementColumn(tblInfo, idxInfo)
-			if err != nil {
-				job.State = model.JobStateCancelled
-				return nil, nil, nil, err
-			}
-		}
-	}
```

Comment:
Please add a test so that it breaks when we support dropping column covered with composite index:

```sql
mysql> create table t (a int auto_increment, b int, c int, key(a, b));
Query OK, 0 rows affected (0.01 sec)

mysql> alter table t drop column b;
ERROR 8200 (HY000): can't drop column b with composite index covered or Primary Key covered now

```

Reply:
In MySQL, the key will change to key(a), the auto_increment will keep.